packages/frontend/.cct/utils/get-template-component-webpack-config.js

```diff
@@ -0,0 +1,110 @@
+import { dirname, join } from "path";
+import webpack from "webpack";
+import MiniCssExtractPlugin from "mini-css-extract-plugin";
+import postcssOptions from "../../postcss.config.js";
+import { fileURLToPath } from "url";
+import { createRequire } from "module";
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+const require = createRequire(import.meta.url);
+const shared = require("@carrot-kpi/frontend/shared-dependencies.json");
+
+export const getTemplateComponentWebpackConfig = (
+    uniqueName,
+    componentPath,
+    i18nPath,
+    globals,
+    outDir
+) => {
+    const devMode = !!!outDir;
+    return {
+        mode: devMode ? "development" : "production",
+        devtool: false,
+        infrastructureLogging: devMode
+            ? {
+                  level: "none",
+              }
+            : undefined,
+        stats: devMode ? "none" : undefined,
+        entry: join(__dirname, "../../src"),
+        output: {
+            clean: true,
+            path: outDir,
```

Comment:
Should we do something like this when the `outDir` is not provided? I don't know if having the path as `undefined` can cause issues.

```ts
output: {
    clean: true,
    ...(!!outDir ? { path: outDir } : {}),
    uniqueName,
}
```

If `outDir` is not provided, `path` will be absent from the object.